As2 v6 template updates mc (#970)
mulesoft/docs-connectors: as2/6.0/modules/ROOT/pages/as2-connector-examples.adoc

= AS2 Examples - Mule 4
:page-aliases: connectors::as2/AS2-connector-examples.adoc

The following examples show how to configure send and receive messages with receipts in Anypoint Connector for AS2 (AS2 Connector):

> * <<configure-send-message,Configure Sending AS2 Messages>>
> * <<configure-mdn-reciept,Configuring MDN Listener for Receiving Receipts>>
> * <<configure-listener-receive,Configuring an AS2 Listener for Receiving Messages>>

Reviewer: ```suggestion
* <<configure-send-message,Configure Sending AS2 Messages>>
* <<configure-mdn-reciept,Configure an AS2 MDN Listener for Receiving Receipts>>
* <<configure-listener-receive,Configure an AS2 Listener for Receiving Messages>>
```

[[configure-send-message]]
== Configure Sending AS2 Messages

This example shows you how to configure an AS2 send message with an MDN request.

> . In Anypoint Studio, place the AS2 connector in your flow.

Reviewer: ```suggestion
. In Anypoint Studio, place AS2 Connector in your flow.
```

|Keystore config - Private key Password| The private key password, if required.
|===
+
. Configure a connection to the partner AS2 endpoint by providing the *Partner Base URL* value.
. If you want to configure other connection details (such as TLS, proxy, basic authentication, and timeouts), do so now.
. Send the message by using either the *Send with Sync MDN* or *Send with Async MDN* operation:

** `payload` is the message content to send.

** `Filename` is the file name of the message to send.

** The `Boolean` required to configure if the process should fail or if the receipt status is not processed.

. If you choose to request an asynchronous receipt, you must configure `MDN Listener`.

[[configure-mdn-reciept]]

> === Configuring MDN Listener for Receiving Receipts

Reviewer: ```suggestion
=== Configure an AS2 MDN Listener for Receiving Receipts
```

> This example shows you how to set up the MDN Listener to receive receipts.

Reviewer: ```suggestion
This example shows you how to set up the *As 2 mdn listener* source to receive receipts.
```

> Place the connector’s *As 2 mdn listener* as a source in your flow as applicable and reference the global configuration defined here:

Reviewer: ```suggestion
. In the *Mule Palette* view, select *AS2 > As 2 mdn listener*.
. Drag *As 2 mdn listener* to the Studio canvas and reference the global configuration defined here:
```

|Keystore config - Private key Password| The private key password if required.
|===

[[configure-listener-receive]]

> === Configuring an AS2 Listener for Receiving Messages

Reviewer: ```suggestion
=== Configure an AS2 Listener for Receiving Messages
```

> This example shows you how to configure an AS2 Listener to receive MDN requests asynchronously.

Reviewer: ```suggestion
This example shows you how to configure an *As 2 listener* source to receive MDN requests asynchronously.
```

> . Place the connector’s *AS2 listener* component as the source in your flow as applicable for your needs and reference the global configuration as defined here.

Reviewer: ```suggestion
. In the *Mule Palette* view, select *AS2 > As 2 listener*.
. Drag *As 2  listener* to the Studio canvas and reference the global configuration defined here:
```
